### DISCUSSION TITLE
the x-model inside a checkbox doesn't work on "clone checkboxes" (checkboxes with the same value)

### DISCUSSION THREAD

**Discussion by joffreypersia** [alpinejs/alpine#3795] [1. Help] [answered]
Hi, 

I have 2 views in my page : a table view and a mosaïc view. Each element has a checkbox with a value of its ID (to have bulk actions). 

Each time the user clicks on a checkbox element in one view (of the same page, just x-show= false/true) it automatically "check" the other checkbox from the other view but it doesn't trigger the x-model. 

My checkbox is like :
```html
<div class="flex rounded-md shadow-sm">
    <input
        {{ $attributes }}
        type="checkbox"
        class="..."
        value="{{ $item[$col['name']] }}"
        x-model="rowSelected"
    />
</div>
```

And the row is like : 
```html
<tr
    x-data="{
        rowSelected: false,
    }"
    {{ $attributes->merge(['class' => '....']) }}
    :class="{
        '!bg-primary-50 dark:!bg-primary-900': rowSelected,
    }"
>
    {{ $slot }}
</tr>
```

I tried many things, but couldn't create a function that can toggle the rowSelected of the "clone" element (the other checkbox).

I tried this function on a click of the checkbox, but I couldn't access to data variable of the parent. 
```js
 syncTableMosaicCheckboxes(id) {
            // Select all checkboxes
            let allCheckboxes = document.querySelectorAll('input[type=checkbox]');

            // Filter checkboxes based on their value attribute
            let itemsSelected = Array.from(allCheckboxes).filter(checkbox => checkbox.value == id);
            console.log(itemsSelected);
            for (let i = 0; i < itemsSelected.length; i++) {
                itemsSelected[i].$data.dataset.rowSelected = true;
            }
        }
```

Do you guys have an idea ? Either sync the checkboxes with the x-model working on the clone, or make the JS function works

Thank you so much ✨

**Comment by ekwoka** [comment, marked as answer]
get rid of `rowSelected`

wrap the parent state to have a `selectedRows: []` and have the x-model go to `selectedRows`.

Should probably give all of them names as well, to make sure browser form behavior acts correctly. All have same name if they impact the same value.

**Reply by joffreypersia** [reply]
Hi, Ekwoka
Thank you for your help ! 

I didn't know x-model would behave in such an interesting way ! it populate the array correctly and remove existing items if they already exists ! 

So far it is good, but still I have a problem to deal with `rowSelected` because I used this Alpine variable to style my row / card if the checkbox was indeed selected. 

```html
<tr
    x-data="{
        rowSelected: false,
    }"
    {{ $attributes->merge(['class' => '....']) }}
    :class="{
        '!bg-primary-50 dark:!bg-primary-900': rowSelected,
    }"
>
    {{ $slot }}
</tr>
```

(also the all is a Livewire component, but I don't think it will change anything because I would like to have all this behavior client side to avoid reload of the table/mosaic view)

Edit : 
I found a trick, I apply simple css, and it does tje job ! 
```css
    tr:has(input:checked) {
        @apply !bg-primary-50 dark:!bg-primary-900;
    }
```

Thank you

**Reply by ekwoka** [reply]
Of not `has`[does not have great support](https://caniuse.com/?search=css%20%3Ahas)

Personally, I'd just not put the styling on the TR, or otherwise pass in a variable that can be used to tell it which value to watch and have `rowSelected` be defined as 
```js
get rowSelect() {
  return this.values.includes(value)
}
```
